spec/html-spec.coffee

```diff
@@ -414,3 +414,98 @@ describe 'HTML grammar', ->
       """
       for line in invalid.split /\n/
         expect(grammar.firstLineRegex.scanner.findNextMatchSync(line)).toBeNull()
+
+  describe "tags", ->
+    it "tokenizes style tags as such", ->
+      lines = grammar.tokenizeLines '<style>'
+      expect(lines[0][0]).toEqual value: '<', scopes: ['text.html.basic', 'meta.tag.style.html', 'punctuation.definition.tag.html']
+      expect(lines[0][1]).toEqual value: 'style', scopes: ['text.html.basic', 'meta.tag.style.html', 'entity.name.tag.style.html']
+      expect(lines[0][2]).toEqual value: '>', scopes: ['text.html.basic', 'meta.tag.style.html', 'punctuation.definition.tag.html']
+
+    it "tokenizes script tags as such", ->
+      lines = grammar.tokenizeLines '<script>'
+      expect(lines[0][0]).toEqual value: '<', scopes: ['text.html.basic', 'meta.tag.script.html', 'punctuation.definition.tag.html']
+      expect(lines[0][1]).toEqual value: 'script', scopes: ['text.html.basic', 'meta.tag.script.html', 'entity.name.tag.script.html']
+      expect(lines[0][2]).toEqual value: '>', scopes: ['text.html.basic', 'meta.tag.script.html', 'punctuation.definition.tag.html']
+
+    it "tokenizes structure tags as such", ->
+      lines = grammar.tokenizeLines '<html>'
+      expect(lines[0][0]).toEqual value: '<', scopes: ['text.html.basic', 'meta.tag.structure.any.html', 'punctuation.definition.tag.html']
+      expect(lines[0][1]).toEqual value: 'html', scopes: ['text.html.basic', 'meta.tag.structure.any.html', 'entity.name.tag.structure.any.html']
+      expect(lines[0][2]).toEqual value: '>', scopes: ['text.html.basic', 'meta.tag.structure.any.html', 'punctuation.definition.tag.html']
+
+    it "tokenizes block tags as such", ->
+      lines = grammar.tokenizeLines '<div>'
+      expect(lines[0][0]).toEqual value: '<', scopes: ['text.html.basic', 'meta.tag.block.any.html', 'punctuation.definition.tag.begin.html']
+      expect(lines[0][1]).toEqual value: 'div', scopes: ['text.html.basic', 'meta.tag.block.any.html', 'entity.name.tag.block.any.html']
+      expect(lines[0][2]).toEqual value: '>', scopes: ['text.html.basic', 'meta.tag.block.any.html', 'punctuation.definition.tag.end.html']
+
+    it "tokenizes inline tags as such", ->
+      lines = grammar.tokenizeLines '<span>'
+      expect(lines[0][0]).toEqual value: '<', scopes: ['text.html.basic', 'meta.tag.inline.any.html', 'punctuation.definition.tag.begin.html']
+      expect(lines[0][1]).toEqual value: 'span', scopes: ['text.html.basic', 'meta.tag.inline.any.html', 'entity.name.tag.inline.any.html']
+      expect(lines[0][2]).toEqual value: '>', scopes: ['text.html.basic', 'meta.tag.inline.any.html', 'punctuation.definition.tag.end.html']
+
+    it "doesn't tokenize XML namespaces as tags if the prefix is a valid style tag", ->
+      lines = grammar.tokenizeLines '<style:foo>'
+      expect(lines[0][1].value).toNotEqual 'style'
+      expect(lines[0][1].scopes).not.toContain 'entity.name.tag.style.html'
```

Comment:
This is dangerous because this spec will continue to pass if the scope is changed to something else and the behavior regresses.  I think your `.toNotEqual` assertion is ok, but this one probably needs to be changed to a positive `.toEqual`.  Ditto for all the others.